crates/levm/src/call_frame.rs

```diff
@@ -1,20 +1,37 @@
-use crate::{memory::Memory, opcodes::Opcode};
-use bytes::Bytes;
-use ethereum_types::{Address, U256};
+use crate::{
+    memory::Memory,
+    opcodes::Opcode,
+    primitives::{Address, Bytes, U256},
+};
 
 #[derive(Debug, Clone, Default)]
 pub struct CallFrame {
-    pub stack: Vec<U256>, // max 1024 in the future
-    pub memory: Memory,
+    pub gas: U256,
     pub pc: usize,
     pub msg_sender: Address,
     pub callee: Address,
     pub bytecode: Bytes,
     pub delegate: Option<Address>,
     pub msg_value: U256,
+    pub stack: Vec<U256>, // max 1024 in the future
+    pub memory: Memory,
+    pub calldata: Bytes,
+    pub return_data: Bytes,
+    // where to store return data of subcall
+    pub return_data_offset: Option<usize>,
+    pub return_data_size: Option<usize>,
 }
 
 impl CallFrame {
+    pub fn new(bytecode: Bytes) -> Self {
+        Self {
+            bytecode,
+            return_data_offset: None,
+            return_data_size: None,
```

Comment:
Aren't these set as `None` in `..Default::default()`?

Reply:
Right, changed, thanks!